### PR TITLE
APP-991 - Add Organization Protos

### PR DESCRIPTION
**Pull request by anjinai**
**20221219_APP-991 - Add Organization Protos**

**Adds new protos for:**

- CreateOrganization
- GetOrganization
- UpdateOrganization
- DeleteOrganization
- ListOrganizationMembers
- CreateOrganizationInvite
- DeleteOrganizationMember
- DeleteOrganizationInvite
- ResendOrganizationInvite

**Merge Order: 1**

### CODE REVIEW

File: proto/viam/app/v1/app.proto [hunk 1]
```diff
@@ -122,10 +153,96 @@ message Organization {
...
 
+message OrganizationInvite {
+  string invite_id = 1;
+  string organization_id = 2;
+  string email = 3;
+  google.protobuf.Timestamp created_on = 4;
+}
+
+message CreateOrganizationRequest {
+  string user_id = 1;
```
Comment (AdamMagaluk) [inline proto/viam/app/v1/app.proto:173]: Remove `user_id`. We can handle adding the user making the API call to the new org.

File: proto/viam/app/v1/app.proto [hunk 2]
```diff
@@ -122,10 +153,96 @@ message Organization {
...
+  string email = 2;
+}
+
+message ResendOrganizationInviteResponse {
+  OrganizationInvite invite = 1;
+}
+
+message DeleteOrganizationMemberRequest {
+  string organization_id = 1;
+  string user_id = 2;
```
Comment (AdamMagaluk) [inline proto/viam/app/v1/app.proto:241]: Remove.

File: proto/viam/app/v1/app.proto [hunk 3]
```diff
@@ -11,8 +11,38 @@ option go_package = "go.viam.com/api/app/v1";
...
+  // Get an organization
+  rpc GetOrganization(GetOrganizationRequest) returns (GetOrganizationResponse);
+
+  // Update an organization
+  rpc UpdateOrganization(UpdateOrganizationRequest) returns (UpdateOrganizationResponse);
+
+  // Delete an organization
+  rpc DeleteOrganization(DeleteOrganizationRequest) returns (DeleteOrganizationResponse);
+
+  // List all members of an organization
```
Comment (AdamMagaluk) [inline proto/viam/app/v1/app.proto:31]: // List all members of an organization -> // List all members of an organization and all invited members to the organization.

File: proto/viam/app/v1/app.proto [hunk 4]
```diff
@@ -122,10 +153,96 @@ message Organization {
   google.protobuf.Timestamp created_on = 3;
 }
 
+message OrganizationMember {
+  string user_id = 1;
```
Comment (AdamMagaluk) [inline proto/viam/app/v1/app.proto:157]: hmm. What does `user_id` mean in this case?  Is this our internal ID? or the email(s)?

**Review by AdamMagaluk — CHANGES_REQUESTED** [review]
Minor updates and q about user_id.

File: proto/viam/app/v1/app.proto (hunk 4 above)
Comment (anjinai) [reply proto/viam/app/v1/app.proto:157]: This is the member's userId, not their email.

Comment (AdamMagaluk) [reply proto/viam/app/v1/app.proto:157]: Wont we need the email(s) in the frontend?

Comment (anjinai) [reply proto/viam/app/v1/app.proto:157]: Indeed, will update

Comment (anjinai) [reply proto/viam/app/v1/app.proto:157]: updated

**Review by AdamMagaluk — APPROVED** [review]

File: proto/viam/app/v1/app.proto [hunk 5]
```diff
@@ -122,10 +153,96 @@ message Organization {
...
+  string email = 2;
+  google.protobuf.Timestamp date_added = 3;
+}
+
 message ListOrganizationsResponse {
   repeated Organization organizations = 1;
 }
 
+message OrganizationInvite {
+  string invite_id = 1;
```
Comment (michaellee1019) [inline proto/viam/app/v1/app.proto:167]: Do we need to expose invite_id? all operations receive `email` + `org_id` and lookup the invite based on that. So not sure if invite_id is useful at all to return.

**Review by michaellee1019 — APPROVED** [review]
One suggestion. We can remove it later if requires discussion with Adam next week.

File: proto/viam/app/v1/app.proto (hunk 5 above)
Comment (DTCurrie) [reply proto/viam/app/v1/app.proto:167]: IIRC from our design talk, we decided to remove the requirement for invite IDs and instead will rely on `email` + `org_id` for uniqueness/lookup. I am ok with removing it.

**Review by DTCurrie — APPROVED** [review]
✅ mod conversation on `invite_id`

**Pull request closed without merging**